patches/@react-navigation+native+6.0.13.patch

```diff
@@ -0,0 +1,383 @@
+           return;
+         }
+-
+         if (index > previousIndex) {
+           const action = getActionFromStateRef.current(state, configRef.current);
+-
+           if (action !== undefined) {
+             try {
+               navigation.dispatch(action);
+             } catch (e) {
+               // Ignore any errors from deep linking.
+               // This could happen in case of malformed links, navigation object not being initialized etc.
+-              console.warn(`An error occurred when trying to handle the link '${path}': ${typeof e === 'object' && e != null && 'message' in e ? // @ts-expect-error: we're already checking for this
++              console.warn(`An error occurred when trying to handle the link '${path}': ${typeof e === 'object' && e != null && 'message' in e ?
++              // @ts-expect-error: we're already checking for this
+               e.message : e}`);
+             }
+           } else {
+@@ -214,79 +204,67 @@ export default function useLinking(ref, _ref) {
+   }, [enabled, history, ref]);
+   React.useEffect(() => {
+     var _ref$current;
+-
+     if (!enabled) {
+       return;
+     }
+-
+     const getPathForRoute = (route, state) => {
+       // If the `route` object contains a `path`, use that path as long as `route.name` and `params` still match
+       // This makes sure that we preserve the original URL for wildcard routes
+       if (route !== null && route !== void 0 && route.path) {
+         const stateForPath = getStateFromPathRef.current(route.path, configRef.current);
+-
+         if (stateForPath) {
+           const focusedRoute = findFocusedRoute(stateForPath);
+-
+           if (focusedRoute && focusedRoute.name === route.name && isEqual(focusedRoute.params, route.params)) {
+             return route.path;
+           }
+         }
+       }
+-
+       return getPathFromStateRef.current(state, configRef.current);
+     };
+-
+     if (ref.current) {
+       // We need to record the current metadata on the first render if they aren't set
+       // This will allow the initial state to be in the history entry
+       const state = ref.current.getRootState();
+-
+       if (state) {
+         const route = findFocusedRoute(state);
+         const path = getPathForRoute(route, state);
+-
+         if (previousStateRef.current === undefined) {
+           previousStateRef.current = state;
+         }
+-
+         history.replace({
+           path,
+           state
+         });
+       }
+     }
+-
+     const onStateChange = async () => {
+       const navigation = ref.current;
+-
+       if (!navigation || !enabled) {
+         return;
+       }
+-
+       const previousState = previousStateRef.current;
+-      const state = navigation.getRootState(); // root state may not available, for example when root navigators switch inside the container
++      const state = navigation.getRootState();
```

Comment:
@adamgrzybowski could we clean up this patch? I see there is lots of the changes just styling or removing comments, which then makes it hard to follow what are the actually important changes.

Additionally, let's ass some good explanation comments to any change we had to do in this patch, I know this patch is mainly to endsure the state is not broken after resizing from mobile to desktop layout, but it woul dbe great to have only required changes in the diff here.